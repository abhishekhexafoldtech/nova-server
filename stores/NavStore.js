import { defineStore } from "pinia";
import bootstrap from "bootstrap/dist/js/bootstrap.min.js";

export const useNavStore = defineStore("NavStore", {
  // state
  state: () => {
    return {
      bootstrap: bootstrap,
      isSidenavCollapsed: false,
      showConfig: false,
      sidenavType: 'bg-white',
      navFixed: false,
      darkMode: false,
      miniNotification: false,
      breadCrumb: "Nova-Gas"
    };
  },

  actions: {
    toggleSidenavCollapsing() {
      this.isSidenavCollapsed = !this.isSidenavCollapsed;
    },
    toggleShowConfig() {
      this.showConfig = !this.showConfig;
    },
    toggleSidebarColor(payload) {
      this.sidebarColor = payload;
    },
    setSidenavType(payload) {
      this.sidenavType = payload;
    },
    toggleDarkMode() {
      this.darkMode = !this.darkMode;
    },
    toggleNavFixed() {
      this.navFixed = !this.navFixed;
    },
    setBreadCrumbs(data) {
      this.breadCrumb = data
    }
  },
});

