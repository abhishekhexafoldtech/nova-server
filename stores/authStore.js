import { defineStore } from 'pinia';
import { getToken, removeToken, setToken } from '@/utils/auth'
import {  getUserInfo, loginByWeb, logoutByWeb } from '@/api/login'

export const useAuthStore = defineStore("auth", {

  state: () => ({
    token: getToken(),
    user_info: {}, // complete user details
    page_refreshing: true,
  }),

  getters: {
    getLoginToken() {
      return this.token
    },
  },

  actions: {
	// Login
	Login(userCredInfo) {
		return new Promise((resolve, reject) => {
			// make login data set
			let data = {
				email: userCredInfo.email,
				phone: userCredInfo.phone,
				password: userCredInfo.password,
			}
			// Lets fire login with info
			loginByWeb(data).then(response => {
				if(response.status == 200) {
					let responseData = response.data
					// console.log("responseData", responseData)
					let go = true
						if (go) {
							// SITE TOKEN
							this.token = responseData.access_token
							this.user_info = responseData.data
							setToken(responseData.access_token)

							const person = { 
									"id": responseData.data.id,
									"first_name": responseData.data.first_name,
									"last_name": responseData.data.last_name,
									"email": responseData.data.email,
									"phone": responseData.data.phone,
								}
							
							// console.log('person', person)
							localStorage.setItem("user", JSON.stringify(person));

							resolve()
						}
					}
						else {
						alert('Something went wrong!')
						reject()
					}
			}).catch(error => {
				reject(error)
			})
		})
	},

	// Logout
	LogOut() {
		return new Promise((resolve, reject) => {
			this.token = ""
			removeToken()
			localStorage.clear();
			resolve()
			// logoutByWeb().then((response) => {
			// 	this.token = ""
			// 	removeToken()
			// 	resolve(response)
			// }).catch(error => {
			// 	console.log(error)
			// 	reject(error)
			// })
			// .finally(() => {
			// })
		})
	},

	// Navbar loaded or refreshed : both cases fired
	pageLoadedORRefreshed() {
		// Get user info
		return new Promise((resolve) => {
			this.page_refreshing = true;
			// Get user info
			// getUserInfo().then((response) => {
			// this.user_info = response.data.data
			// 	resolve(response);
			// })
			// Get user info from local storage
			let temp_data = localStorage.getItem("user")
			this.user_info = JSON.parse(temp_data)
		});
	},
  },
});
