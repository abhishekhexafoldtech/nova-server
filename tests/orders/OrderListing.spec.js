import { describe, it, expect } from "vitest";
import { shallowMount } from "@vue/test-utils";
import OrderListing from "~/pages/orders/index.vue";
import { pinia } from "@/tests/mockpinia";
import { useNavStore } from "@/stores/NavStore";

describe("OrderListing", () => {
  const navStore = useNavStore(pinia);
  it("checked table is rendered", () => {
    const wrapper = shallowMount(OrderListing);
    const singleFileUpload = wrapper.findComponent({ name: "Table" });
    expect(singleFileUpload.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});