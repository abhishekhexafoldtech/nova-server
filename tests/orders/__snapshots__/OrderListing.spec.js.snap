// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OrderListing > checked table is rendered 1`] = `
"<section>
  <!-- Display an empty state if there are no orders -->
  <!--v-if-->
  <!-- Display the orders table if data is present -->
  <div class=\\"container-area\\">
    <h4 class=\\"page_heading mb-4\\">Orders</h4>
    <div class=\\"table-area\\">
      <table-stub tableconfig=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" tabledata=\\"\\" tablequery=\\"[object Object]\\" tableheading=\\"\\" headingrowreverse=\\"true\\" tableloadingstatus=\\"false\\" tablesubheading=\\"\\" tablesearchvisibility=\\"true\\" actionvisibility=\\"false\\" viewbuttonvisibility=\\"false\\" editbuttonvisibility=\\"true\\" deletebuttonvisibility=\\"true\\" moreactionsvisibility=\\"false\\" resetpasswordvisibility=\\"true\\" approverequestsvisibility=\\"false\\" declinerequestsvisibility=\\"false\\" viewonboardingformvisibility=\\"false\\" roleandpermissionvisibility=\\"true\\" tablecheckboxvisibility=\\"true\\" export=\\"true\\" refresh=\\"true\\" filter=\\"true\\" filterdata=\\"[object Object],[object Object],[object Object],[object Object]\\" downloadvisibility=\\"false\\" downloadlinkcontent=\\"\\" style=\\"border-radius: 20px;\\" addbuttonvisibility=\\"false\\"></table-stub>
    </div>
  </div>
</section>"
`;
